components/ws-manager/pkg/manager/manager_ee.go

```diff
@@ -54,21 +56,66 @@ func (m *Manager) TakeSnapshot(ctx context.Context, req *api.TakeSnapshotRequest
 		return nil, status.Errorf(codes.FailedPrecondition, "can only take snapshots of running workspaces")
 	}
 
-	sync, err := m.connectToWorkspaceDaemon(ctx, workspaceObjects{Pod: pod})
-	if err != nil {
-		return nil, status.Errorf(codes.Unavailable, "cannot connect to workspace daemon: %q", err)
+	_, pvcFeatureEnabled := wso.Pod.Labels[pvcWorkspaceFeatureLabel]
+	pvcVolumeSnapshotClassName := ""
+
+	if _, ok := wso.Pod.Labels[workspaceClassLabel]; ok {
+		wsClassName := wso.Pod.Labels[workspaceClassLabel]
+
+		workspaceClass := m.Config.WorkspaceClasses[wsClassName]
+		if workspaceClass != nil {
+			pvcVolumeSnapshotClassName = workspaceClass.PVC.SnapshotClass
+		}
+	}
+
+	if pvcFeatureEnabled {
+		workspaceID, _ := wso.WorkspaceID()
```

Comment:
Should we check the workspace ID exists before using it?
For example
```go
id, ok := wso.WorkspaceID()
if !ok {
    ....
}
```

Reply:
`sts, err := m.getWorkspaceStatus(*wso)` on line 50 above checks for that already, so not necessary to do it again.